pkg/chunk/aws/metrics_autoscaling.go

```diff
@@ -137,6 +172,21 @@ func (m metricsData) computeScaleUp(current, expected chunk.TableDesc) int64 {
 	return scaleUp
 }
 
+func (m metricsData) computeReadScaleDown(current, expected chunk.TableDesc) int64 {
+	usageRate := m.usageReadRates[expected.Name]
+	return int64(usageRate * 100.0 / expected.ReadScale.TargetValue)
+}
+
+func (m metricsData) computeReadScaleUp(current, expected chunk.TableDesc) int64 {
+	scaleUp := int64(float64(current.ProvisionedRead) * m.cfg.ScaleUpFactor)
```

Comment:
I'm going to revisit these functions once I get the AWS errors covered up (and before I remove the WIP tag ), because I agree there is a high level of overlap. I just wanted to get some good tests passes clearing before I come and break the code again :)

Reply:
Took a pass at refactoring these computeX and scaleX functions to reduce the duplication. This is the latest code push